sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -283,6 +283,79 @@ export class ManagementClient extends LinkEntity {
       : undefined;
   }
 
+  // TODO - Any better name??
+  private async _acquireLockHelper(
+    message_id: string | number | Buffer | undefined,
+    timeoutInMs: number
+  ): Promise<number> {
+    const initOperationStartTime = Date.now();
+    if (!this._isMgmtRequestResponseLinkOpen()) {
+      log.mgmt(
+        "[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId
+      );
+
+      const actionAfterTimeout = () => {
+        const desc: string = `The request with message_id "${message_id}" timed out. Please try again later.`;
+        const e: Error = {
+          name: "OperationTimeoutError",
+          message: desc
+        };
+
+        throw e;
+      };
+
+      const waitTimer = setTimeout(actionAfterTimeout, timeoutInMs);
+
+      try {
+        await defaultLock.acquire(this.managementLock, () => {
+          return this._init();
+        });
+      } catch (err) {
+        throw err;
+      } finally {
+        clearTimeout(waitTimer);
+      }
+    }
+    // Returns the time taken by the init operation
+    return Date.now() - initOperationStartTime;
+  }
+
+  // TODO - Any better name??
+  private async _managementLinkSendRequestHelper(
+    request: AmqpMessage,
+    sendRequestOptions: SendRequestOptions
+  ): Promise<AmqpMessage> {
+    try {
+      return await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions);
+    } catch (err) {
+      err = translate(err);
+      const address =
+        this._mgmtReqResLink || this._mgmtReqResLink!.sender.address
+          ? "address"
+          : this._mgmtReqResLink!.sender.address;
+      log.warning(
+        "[%s] An error occurred during send on management request-response link with address " +
+          "'%s': %O",
+        this._context.namespace.connectionId,
+        address,
+        err
+      );
+      throw err;
+    }
+  }
+
+  private async _acquireLockAndSendRequest(
+    request: AmqpMessage,
+    retryTimeoutInMs: number,
+    sendRequestOptions: SendRequestOptions
+  ): Promise<AmqpMessage> {
+    const timeTakenByInit = await this._acquireLockHelper(request.message_id, retryTimeoutInMs);
+    // Left over time - remainingOperationTimeoutInMs
+    sendRequestOptions.timeoutInMs = retryTimeoutInMs - timeTakenByInit;
+    return this._managementLinkSendRequestHelper(request, sendRequestOptions);
```

Comment:
We are introducing 3 new private methods. I don't see the benefit of splitting them up instead of having them in a single function. I suggest the name `_makeManagementRequest()` which takes care of
- If link doesnt exist, acquire lock and call init() within the timer
- call `sendRequest()` on the management request response link

I wouldn't recommend having "acquire lock" in the name as acquiring the lock is not the main operation. The main thing is to send the request. To send the request, we have to create link if it doesnt exist. To create the link, we need lock. So acquiring lock is more of an implementation detail rather than the main show

Reply:
Okay, merged all three methods into one single `_makeManagementRequest`.

Reply:
Thanks! Looks much better and easy to read